spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DiscoveryClientServiceInstanceListSupplier.java

```diff
@@ -41,20 +41,21 @@
 
 	private final String serviceId;
 
-	private final Flux<ServiceInstance> serviceInstances;
+	private final Flux<List<ServiceInstance>> serviceInstances;
 
 	public DiscoveryClientServiceInstanceListSupplier(DiscoveryClient delegate,
 			Environment environment) {
 		this.serviceId = environment.getProperty(PROPERTY_NAME);
 		this.serviceInstances = Flux
-				.defer(() -> Flux.fromIterable(delegate.getInstances(serviceId)))
-				.subscribeOn(Schedulers.boundedElastic());
+				.defer(() -> Flux.fromIterable(delegate.getInstances(serviceId))
```

Comment:
@OlgaMaciaszek  
I see here two problems: 
1. Blocking `DiscoveryClient.getInstances` executes before switch to `Schedulers.boundedElastic()`.    `.subscribeOn(Schedulers.boundedElastic())` should be applied to `Flux.defer()` - IMO this is critical  
2. `Flux.fromIterable(delegate.getInstances(serviceId)).collectList().flux()` could be simplified to `Flux.defer(() -> Flux.just(delegate.getInstances(serviceId)))`  

so final result could look like 
```java
this.serviceInstances = Flux
		.defer(() -> Flux.just(delegate.getInstances(serviceId)))
		.subscribeOn(Schedulers.boundedElastic());
```

Reply:
You're right. Thanks. Fixing it.